nilearn/decomposition/multi_pca.py

```diff
@@ -359,3 +383,102 @@ def inverse_transform(self, component_signals):
         # XXX: dealing properly with 2D/ list of 2D data?
         return [nifti_maps_masker.inverse_transform(signal)
                 for signal in component_signals]
+
+    def _get_filter_and_mask_parameters(self):
+        parameters = get_params(MultiNiftiMasker, self)
+        # Remove non specific and redudent parameters
+        for param_name in ['memory', 'memory_level', 'confounds',
+                           'verbose', 'n_jobs']:
+            parameters.pop(param_name, None)
+
+        parameters['detrend'] = True
+        return parameters
+
+    def _sort_components(self, imgs, confounds=None):
+        """ Sort components by score obtained on test set imgs
+        """
+        score = self.score(imgs, confounds, per_component=True).mean(axis=0)
+        self.score_ = -np.sort(-score)
```

Comment:
I am guessing you do that to sort in the descending order.

You can use `np.sort(score)[::-1]` instead.

You could compute once the argsort and reuse in this line and the line below:

```python
index = np.argsort(score)[::-1]
self.score_ = score[index]
self.components_ = self.components_[index]
```